bin/db-initialization.py

```diff
@@ -1,87 +1,101 @@
-
-        await conn.execute(q, *p)
-
-        # Create Caboose Role
-
-        q, p = render(
-            """
-                CREATE ROLE :un WITH LOGIN IN ROLE :ir PASSWORD ':pw';
-            """,
-            un=V(os.environ["CABOOSE_ROLE_USER"]),
-            pw=V(os.environ["CABOOSE_ROLE_PASS"]),
-            ir=V(owner_role_un),
-        )
-
-        await conn.execute(q, *p)
-
-        # Create Conductor Role
-
-        q, p = render(
-            """
-                CREATE ROLE :un WITH LOGIN IN ROLE :ir PASSWORD ':pw';
-            """,
-            un=V(os.environ["CONDUCTOR_ROLE_USER"]),
-            pw=V(os.environ["CONDUCTOR_ROLE_PASS"]),
-            ir=V(owner_role_un),
-        )
-
-        await conn.execute(q, *p)
-
-        # Create database with owner role as owner
-
-        q, p = render(
-            """
-                CREATE DATABASE :db WITH OWNER :dbo;
-            """,
-            db=V(DB_NAME),
-            dbo=V(owner_role_un),
-        )
-
-        await conn.execute(q, *p)
-
-        await conn.close()
-
-    asyncio.run(setup())
+import sys
+
+import asyncpg
+from buildpg import V, render
+
+from swoop.db import SwoopDB
+
+OWNER_ROLE_NAME = "swoop"
+APPLICATION_ROLES: list[str] = [
+    "API_ROLE",
+    "CABOOSE_ROLE",
+    "CONDUCTOR_ROLE",
+    "MIGRATION_ROLE",
+]
+
+
+def stderr(*args, **kwargs) -> None:
+    kwargs["file"] = sys.stderr
+    print(*args, **kwargs)
+
+
+async def create_owner_role(conn: asyncpg.Connection, owner_role_name: str) -> None:
+    q, p = render(
+        "CREATE ROLE :un",
+        un=V(owner_role_name),
+    )
+    await conn.execute(q, *p)
+
+
+async def create_application_role(
+    conn: asyncpg.Connection,
+    role: str,
+    owner_role_name: str,
+) -> None:
+    q, p = render(
+        "CREATE ROLE :un WITH LOGIN IN ROLE :ir PASSWORD ':pw'",
+        un=V(os.environ[f"{role}_USER"]),
+        pw=V(os.environ[f"{role}_PASS"]),
+        ir=V(owner_role_name),
+    )
+    await conn.execute(q, *p)
+
+
+async def create_database(
+    conn: asyncpg.Connection,
+    dbname: str,
+    owner_role_name: str,
+) -> None:
+    q, p = render(
+        "CREATE DATABASE :db WITH OWNER :dbo",
+        db=V(dbname),
+        dbo=V(owner_role_name),
+    )
+    await conn.execute(q, *p)
+
+
+async def db_initialization() -> None:
+    dbname: str = os.environ["PGDATABASE"]
+
+    async with SwoopDB.get_db_connection(database="") as conn:
+        try:
+            stderr(f"Creating owner role '{OWNER_ROLE_NAME}'...")
+            await create_owner_role(conn, OWNER_ROLE_NAME)
+        except asyncpg.DuplicateObjectError:
+            stderr("Owner already exists, skipping.")
+
+        for role in APPLICATION_ROLES:
+            try:
+                stderr(f"Creating application role {role}...")
+                await create_application_role(conn, role, OWNER_ROLE_NAME)
+            except asyncpg.DuplicateObjectError:
+                stderr("Role already exists, skipping.")
```

Comment:
For these roles, I am uncertain if they should be updated on existence. I would lean to not doing so.

One argument for updating them would be that an update would allow us to roll passwords by running this script. But I would suggest that rolling role passwords should be a separate, explicit operation (which we should support, but isn't an urgent need), and that running this script should only be for the case where the user is trying to create the database.